src/controllers/accommodationController.js

```diff
@@ -0,0 +1,82 @@
+/* eslint-disable max-len */
+import Model from '../models';
+import * as accommodationServices from '../services/accommodationServices';
+import { respondWithSuccess, respondWithWarning } from '../helpers/responseHandler';
+
+const { Accommodation, Room } = Model;
+/**
+ * Function creates accommodation
+ * @param {object} req
+ * @param {object} res
+ * @returns {object} response object
+ */
+export const createAccommodation = async (req, res) => {
+  try {
+    req.accommodation = req.body;
+    req.accommodation.userId = req.auth.id;
+    req.accommodation.images = req.files.map(file => file.secure_url);
+    const accommodation = await accommodationServices.create(Accommodation, req.accommodation);
```

Comment:
We can refactor this better
```
const whatever = {
     ...req.body,
     userId: req.auth.id,
     images: req.files.map(file => file.secure_url),
}

const accommodation = await accommodationServices.create(Accommodation, whatever);
```

Reply:
Makes sense. I have a feeling that this implementation is bad. But, I couldn't wrap my head around any solution. Thanks for showing me a way out.